live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/mapper/property/PropertyMappingHelper.java

```diff
@@ -148,6 +165,30 @@
         return time == null ? null : time.toInstant().toEpochMilli();
     }
 
+    public static boolean shouldNotUpdateContentField(RepoEvent<DataAttributes<NodeResource>> event)
+    {
+        Function<NodeResource, Serializable> nameGetter = NodeResource::getName;
```

Comment:
Any views on this one?

Reply:
Looks sensible to me. These `Function` variables could be instantiated after the conditional. This way you can fail/return fast without creating any variables. WDYT?
```suggestion
        if (isEventTypeCreated(event))
        {
            return false;
        }
         Function<NodeResource, Serializable> nameGetter = NodeResource::getName;
         Function<NodeResource, Optional<ContentInfo>> contentInfoGetter = resource -> ofNullable(resource.getContent());
        Function<NodeResource, Serializable> mimeTypeGetter = resource -> contentInfoGetter.apply(resource).map(ContentInfo::getMimeType).orElse(null);
        Function<NodeResource, Serializable> sizeGetter = resource -> contentInfoGetter.apply(resource).map(ContentInfo::getSizeInBytes).orElse(null);
```

Reply:
Yep, thank you - I agree. I ended up having to change this logic anyway because the logic for including the content metadata section is more complex. We only want to include it if:
* The MIME type is new or has been updated
* OR The file size is new or has been updated
* OR The name has been updated _and_ there is content (i.e. a MIME type or file size are present in the latest node resource section).

In particular we don't want to include it when a folder's name is updated (since there is no content).

The latest commit has this fix, as well as the code needed to make the bulk ingestion work in line with the rest of this code.